plugins/sign-on/openid-connect-bcc-customizations/includes/snippets.php

```diff
@@ -1,37 +1,56 @@
 <?php
+  /**
+   * Redirect user with only read access to home_url when accessing the admin dashboard.
+   */
+  add_action('admin_init', function() {
+    if ( !is_user_logged_in() ) {
+      return null;
+    }
+
+    $user_meta = get_userdata( get_current_user_id() );
+    $only_read_access = true;
+
+    foreach ($user_meta->allcaps as $cap => $value) {
+      if ( $cap != 'read' && ! in_array($cap, $user_meta->roles) ) {
+        $only_read_access = false;
+      }
+    }
+
+    if ( $only_read_access && !defined('DOING_AJAX') ) {
+      wp_redirect(home_url());
+      exit;
+    }
+  });
+
   /**
    * Update user's email if it was changed in PMO.
    */
   add_action('openid-connect-generic-update-user-using-current-claim', function($user, $user_claim) {
-    if ($user_claim['email'] != $user->user_email) {
-      $args = array(
+    if ( $user_claim['email'] != $user->user_email ) {
```

Comment:
Also check that email isn't blank, before overwriting current email. Not sure if this feature is strictly required if we're not using email to authenticate.

Reply:
This snippet is actually not new, and even if we don't use the email to authenticate, it is useful at least to have consistent data with Members as well as not create a duplicate account which, as far as I remember, it happened in a previous project.
Good suggestions though, I add that.

Reply:
Does this code also run on the shared/generated user?

Reply:
No. We don't want that.